src/shaders/terrain_raster.vertex.glsl

```diff
@@ -7,6 +7,7 @@ attribute vec2 a_texture_pos;
 varying vec2 v_pos0;
 
 #ifdef FOG
+uniform mat4 u_cam_matrix;
```

Comment:
Do you think this should be directly put in the _prelude like the other fog specific uniforms or is that the only place we'd use this matrix? https://github.com/mapbox/mapbox-gl-js/blob/fog-implementation/src/shaders/_prelude_fog.fragment.glsl#L3_L6

Reply:
Yep that makes sense, the varying `v_depth` and any fog related declaration should as well the, right?

Reply:
I think we could yes, I haven't put it there and it's duplicated but we can clean that to put in the prelude, and may wrap that transform function similarly to how fog is [applied](https://github.com/mapbox/mapbox-gl-js/blob/fog-implementation/src/shaders/_prelude_fog.fragment.glsl#L17) (so changing it globally is easier)